assets/components/src/web-preview/index.js

```diff
@@ -30,95 +30,152 @@ class WebPreview extends Component {
 	state = {
 		device: 'desktop',
 		loaded: false,
-		previewVisibility: false,
+		isPreviewVisible: false,
+		modalDOMElement: null,
 	};
 
+	/**
+	 * Create a div and append it to the body, so that the modal is on the top layer
+	 */
+	componentDidMount() {
+		const modalDOMElement = document.createElement( 'div' );
```

Comment:
The [React Portals doc](https://reactjs.org/docs/portals.html) suggests creating the element in the constructor and adding to the DOM in `componentDidMount`. It seems like your approach is going to be effectively identical, but I'm wondering if the official approach should be adhered to here?

Reply:
I don't know what would be the practical difference. But when introducing use of single div ([comment below](https://github.com/Automattic/newspack-plugin/pull/404#discussion_r370108298)) I've noticed that if one instance should be aware of the div mounted by the previous one, it has to happen in `componentDidMount`